src/hash.h

```diff
@@ -132,20 +132,33 @@ class CHashWriter
         ctx.Write((const unsigned char*)pch, size);
     }
 
-    // invalidates the object
+    /** Compute the double-SHA256 hash of all data written to this object.
+     *
+     * Invalidates this object.
+     */
     uint256 GetHash() {
         uint256 result;
-        ctx.Finalize((unsigned char*)&result);
+        ctx.Finalize(result.begin());
+        ctx.Reset().Write(result.begin(), CSHA256::OUTPUT_SIZE).Finalize(result.begin());
+        return result;
+    }
+
+    /** Compute the SHA256 hash of all data written to this object.
+     *
+     * Invalidates this object.
+     */
+    uint256 GetSHA256() {
+        uint256 result;
+        ctx.Finalize(result.begin());
         return result;
     }
 
     /**
      * Returns the first 64 bits from the resulting hash.
      */
     inline uint64_t GetCheapHash() {
-        unsigned char result[CHash256::OUTPUT_SIZE];
-        ctx.Finalize(result);
-        return ReadLE64(result);
+        uint256 result = GetHash();
```

Comment:
Shouldn't `GetCheapHash` be calling `GetSHA256`? I.e. "Cheap" refers to a single rather than a double hash. cc @JeremyRubin @sipa

Reply:
No because it's a behavior change :/

If someone wants to make sure that we don't rely on cheaphashes being consistent across boots then yes.

Reply:
Addrman's bucketing relies on it.